From d9554019340177a2fb0d033fc6ffcf45f025cb2f Mon Sep 17 00:00:00 2001
From: "James R. Barlow" <james@purplerock.ca>
Date: Wed, 6 Jan 2021 00:22:16 -0800
Subject: [PATCH 3/4] libqpdf 10.1.0 raises different exception

The different errors are acceptable to us; actually they are more
correct than the original behavior.

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 tests/test_filters.py | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/tests/test_filters.py b/tests/test_filters.py
index b394795..33124bf 100644
--- a/tests/test_filters.py
+++ b/tests/test_filters.py
@@ -74,19 +74,18 @@ class FilterInvalid(pikepdf.TokenFilter):
 
 def test_invalid_handle_token(pal):
     page = pikepdf.Page(pal.pages[0])
-    with pytest.raises(pikepdf.PdfError):
-        result = page.get_filtered_contents(FilterInvalid())
+    with pytest.raises((TypeError, pikepdf.PdfError)):
+        page.get_filtered_contents(FilterInvalid())
 
 
 def test_invalid_tokenfilter(pal):
     page = pikepdf.Page(pal.pages[0])
     with pytest.raises(TypeError):
-        result = page.get_filtered_contents(list())
+        page.get_filtered_contents(list())
 
 
 def test_tokenfilter_is_abstract(pal):
     page = pikepdf.Page(pal.pages[0])
-    try:
-        result = page.get_filtered_contents(pikepdf.TokenFilter())
-    except pikepdf.PdfError:
-        assert 'Tried to call pure virtual' in pal.get_warnings()[0]
+    with pytest.raises((RuntimeError, pikepdf.PdfError)):
+        page.get_filtered_contents(pikepdf.TokenFilter())
+    assert 'Tried to call pure virtual' in pal.get_warnings()[0]
-- 
2.29.2

